docs/src/content/docs/reference/configuration.mdx

```diff
@@ -514,6 +514,13 @@ Sets the delimiter between page title and site title in the page’s `<title>` t
 By default, every page has a `<title>` of `Page Title | Site Title`.
 For example, this page is titled “Configuration Reference” and this site is titled “Starlight”, so the `<title>` for this page is “Configuration Reference | Starlight”.
 
+### `disable404Route`
+
+**type:** `boolean`  
+**default:** `'false'`
+
+Disables injecting Starlight's default [404 page](https://docs.astro.build/en/core-concepts/astro-pages/#custom-404-error-page) so it can be overwritten with a custom one.
```

Comment:
Suggesting a minor tweak here to make the second part more “user desire => action”.

```suggestion
Disables injecting Starlight's default [404 page](https://docs.astro.build/en/core-concepts/astro-pages/#custom-404-error-page). To use a custom `src/pages/404.astro` route in your project, set this option to `false`.
```